kit/src/elements/select/mod.rs

```diff
@@ -18,10 +20,17 @@ pub fn emit(cx: &Scope<Props>, s: String) {
     }
 }
 
+fn remove_duplicates(values: Vec<String>) -> Vec<String> {
```

Comment:
You could do this

```rust
fn remove_duplicates(values: Vec<String>) -> Vec<String> {
    let unique_values = HashSet::<_>::from_iter(values);
    Vec::from_iter(unique_values)
}

```

Or if you prefer, could do

```rust
fn remove_duplicates<I>(values: I) -> Vec<String>
where
    I: IntoIterator<Item = String>,
{
    let unique_values = HashSet::<_>::from_iter(values);
    Vec::from_iter(unique_values)
}
```

Though the first one might be more preferable but either way it would reduce the need of `cloned` since you can pass the iterator (or values) directly into `from_iter` and convert it back into a `Vec` from there, which should remove any duplicate values. 

Note: Using this method will change the order of the array since `HashSet` is ordered based on hash of the value, so keep that in mind if the order of the array matters. If it doesnt, then either or of these functions will due.

Reply:
I would have provided an example using the reference instead, but that would have also gotten cloned internally either way so I left it like this

Reply:
Just as another precaution, it might be best to check for any values before pushing into a vector a head of time

Reply:
> > Just as another precaution, it might be best to check for any values before pushing into a vector a head of time
> 
> What do you mean exactly here @dariusc93 ?

Where ever `Select` is being used at, it would be best to check the supplied vector there before passing it to `Select`. As an example, in `GeneralSettings`, themes are being supplied to `option`. Moving up into `get_available_themes`, it would be best to check the vector for any themes that may be there. Eg 

```rust
if !themes.contains(&theme) {
   themes.push(theme);
}
```
Although, because names are used in `GeneralSettings`, it would be best to manually implement `PartialEq` and checking to see if the name field equal each other. eg

```rust
#[derive(Clone, Debug, Default, Deserialize, Serialize)]
pub struct Theme {
    pub filename: String,
    pub name: String,
    pub styles: String,
}

impl PartialEq for Theme {
    fn eq(&self, other: &Self) -> bool {
        self.name.eq(&other.name)
    }
}
```

That way, when it checks to see if another `Theme` equals another, it would compare the name instead of the whole struct. 

Same with font too since the name is used there apart of the settings.

Reply:
Does that PR supersede this one?